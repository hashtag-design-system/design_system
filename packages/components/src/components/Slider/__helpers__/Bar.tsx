import React from "react";
import { SliderFProps } from "../index";

export const Bar = React.forwardRef<HTMLSpanElement, Pick<SliderFProps, "className">>(({ className }, ref) => {
  return (
    <span className={`slider__bar shadow__inset-sm ${className}`}>
      <span ref={ref} className={`slider__bar__progress ${className}`} />
    </span>
  );
});
