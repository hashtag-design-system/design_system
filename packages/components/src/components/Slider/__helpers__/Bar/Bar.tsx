import React from "react";
import { SliderFProps } from "../../index";

export const Bar = React.forwardRef<HTMLSpanElement, Pick<SliderFProps, "className">>(({ className }, ref) => {
  return (
    <span className={`slider__bar shadow__inset-sm ${className}`} data-testid="slider-bar">
      <span ref={ref} className={`slider__bar__progress ${className}`} />
    </span>
  );
});
