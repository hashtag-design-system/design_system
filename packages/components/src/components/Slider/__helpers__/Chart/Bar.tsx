import React from "react";

type Props = Pick<React.CSSProperties, "height">;

export const Bar: React.FunctionComponent<Props> = ({ height = 50 }) => {
  return (
    <div className="slider__chart__bar" style={{ height: `${height}%` }} data-testid="slider-chart-bar"></div>
  );
};
