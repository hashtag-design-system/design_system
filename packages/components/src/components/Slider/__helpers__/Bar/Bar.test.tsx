import { render, screen } from "@testing-library/react";
import { Bar } from "./Bar";

describe("Slider <Bar />", () => {
  test("default behaviour", () => {
    render(<Bar />);
    const bar = screen.getByTestId("slider-bar");

    expect(bar).toBeVisible();
    expect(bar).toHaveAttribute("class");
    expect(bar).toMatchSnapshot();
    expect(bar.style.width).toBeDefined();
    expect(bar.children.length).toBe(1);
    expect(bar.children[0]).toHaveAttribute("class");
  });
  test("custom className", () => {
    render(<Bar className="custom-className" />);
    const bar = screen.getByTestId("slider-bar");

    expect(bar).toHaveClass("custom-className");
    expect(bar.children[0]).toHaveClass("custom-className");
  });
});
