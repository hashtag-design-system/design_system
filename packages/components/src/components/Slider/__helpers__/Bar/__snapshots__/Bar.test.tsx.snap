// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider <Bar /> default behaviour 1`] = `
<span
  class="slider__bar shadow__inset-sm "
  data-testid="slider-bar"
>
  <span
    class="slider__bar__progress undefined"
  />
</span>
`;
